.github/workflows/ci-pullrequest-main.yml

```diff
@@ -0,0 +1,73 @@
+name: Mock Classifier CI Pipeline
+
+on:
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  BuildTest:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v2
+      with:
+        dotnet-version: 6.0.x
+        
+    - name: Dotnet Restore Tools
+      run: dotnet tool restore --tool-manifest "src/.config/dotnet-tools.json"
+        
+    - name: Dotnet Restore
+      run: dotnet restore src/*.sln
+        
+    - name: Dotnet Format
+      run: dotnet format src/*.sln --no-restore
+      
+    - name: Dotnet Build
+      run: dotnet build src/*.sln --no-restore
+     
+    - name: Dotnet Test (unit tests only)
+      run: dotnet test src/*.UnitTests/*.csproj --no-build --verbosity normal --collect:"XPlat Code Coverage"
+      
+    - name: Copy Coverage output to root
+      run: cp src/*.UnitTests/TestResults/**/coverage.cobertura.xml coverage.cobertura.xml
+      
+    - name: Code Coverage Summary
+      uses: irongut/CodeCoverageSummary@v1.2.0
+      with:
+        filename: coverage.cobertura.xml
+        fail_below_min: true
+        format: markdown
+        output: both
+        thresholds: '80'
+    
+    - name: Upload Coverage Markdown Report
+      uses: actions/upload-artifact@v3.0.0
+      with:
+        name: coverage-report-markdown
+        path: code-coverage-results.md
```

Comment:
Just curious - do we need both Xml and MD?

Reply:
Not sure but i thought it would be worth adding them both for now just until we work out how they will be used. The XML is more detailed but the HTML is more readable.